ome/datasource_firmware_catalog_test.go

```diff
@@ -0,0 +1,117 @@
+limitations under the License.
+*/
+
+package ome
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"terraform-provider-ome/helper"
+	"terraform-provider-ome/utils"
+	"testing"
+
+	. "github.com/bytedance/mockey"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestDataSource_FirmwareCatalogRead(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// No filter so should return all values
+			{
+				Config: firmwareCatAll + outputs,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckOutput("fetched_any", "true"),
+					resource.TestCheckOutput("fetched_many", "true"),
+					resource.TestCheckOutput("fetched_one", "false"),
+				),
+			},
+			// Empty Filter so should return all values
+			{
+				Config: firmwareCatAllEmptyNameFilter + outputs,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckOutput("fetched_any", "true"),
+					resource.TestCheckOutput("fetched_many", "true"),
+					resource.TestCheckOutput("fetched_one", "false"),
+				),
+			},
+			// Using the name filter
+			{
+				Config: firmwareCatFilter + outputs,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckOutput("fetched_any", "true"),
+					resource.TestCheckOutput("fetched_many", "false"),
+					resource.TestCheckOutput("fetched_one", "true"),
+				),
+			},
+			// Error getting firmware catalogs
+			{
+				PreConfig: func() {
+					FunctionMocker = Mock(helper.GetAllCatalogFirmware).Return(nil, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      firmwareCatAll,
+				ExpectError: regexp.MustCompile(`.*Error fetching firmware catalogs*.`),
+			},
+			// Processing error test
+			{
+				PreConfig: func() {
+					if FunctionMocker != nil {
+						FunctionMocker.UnPatch()
+					}
+					FunctionMocker = Mock(utils.CopyFields).Return(fmt.Errorf("Mock error")).Build()
+				},
+				Config:      firmwareCatAll,
+				ExpectError: regexp.MustCompile(`.*Error processing firmware catalogs*.`),
+			},
+		},
+	})
+}
+
+var outputs = `
+output "fetched_many" {
+	value = length(data.ome_firmware_catalog.data-catalog.firmware_catalogs) > 1
+}
+  
+output "fetched_any" {
+	value = length(data.ome_firmware_catalog.data-catalog.firmware_catalogs) != 0
+}
+
+output "fetched_one" {
+	value = length(data.ome_firmware_catalog.data-catalog.firmware_catalogs) == 1
+}
+`
+
+var firmwareCatAll = testProvider + `
+data "ome_firmware_catalog" "data-catalog" {
+}
+`
+
+var firmwareCatAllEmptyNameFilter = testProvider + `
+data "ome_firmware_catalog" "data-catalog" {
+	names = []
+}  
+`
+
+var firmwareCatFilter = testProvider + `
+
+data "ome_firmware_catalog" "data-catalog" {
+	names = ["tfacc_catalog_dell_online_1"]
```

Comment:
please make the catalog name configurable through environt vars

Reply:
Updated